### PR TITLE
Add synthetic monitoring

### PR DESCRIPTION
**Pull request by rddefauw**
This PR adds a CloudWatch canary to the app stack in each region.  

The canary checks to make sure that we can log in to the app and start a game.  A CloudWatch alarm triggers if the canary success rate falls below 95%.

We use the canary as a readiness check in each cell, and added a note in the README for other possible uses.

Note that in order to use the higher level CDK constructs for the Canary, I had to revert to CDK v1.

### CODE REVIEW

**Comment by sebsto** [comment]
Thanks for this PR.  Is there a way to add the same without "downgrading" to CDK v1 ?  I consciously used CDK v2 to resist the test of time :-)

**Comment by rddefauw** [comment]
Unfortunately not easy to use v2, as it doesn't yet support the Canary construct for Synthetics.

**Comment by sebsto** [comment]
I would suggest to wait for support in v2 to merge in the main branch.  OK for merging in a v1 branch ?

**Comment by rddefauw** [comment]
Sounds good.

File: lib/tictactoe-app-cdk-stack.ts [hunk 1]
```diff
@@ -49,7 +48,7 @@ export class TictactoeAppCdkStack extends Stack {
       assumedBy: new iam.ServicePrincipal('ec2.amazonaws.com')
     });
     // allow the role to read / write on the table
-    props?.table.grantReadWriteData(dynamoDBRole);
+    props?.table.grantFullAccess(dynamoDBRole);
```
Comment (sebsto) [inline lib/tictactoe-app-cdk-stack.ts]: Why a full access is needed ? We try to keep minimum permissions on the samples.

**Review by sebsto — COMMENTED** [review]
Can you also consider not including the lib/node_modules in the repo ?  This should be downloaded / installed by the developer during the `npm install` no ?

**Comment by rddefauw** [comment]
I am not including any node modules.  The only content under that folder is the canary script (canary.js) which I wrote.  The folder structured is mandated by the CDK modules for CloudWatch Synthetics.

File: lib/tictactoe-app-cdk-stack.ts (hunk 1 above)
Comment (rddefauw) [reply lib/tictactoe-app-cdk-stack.ts]: The `grantReadWriteData` doesn't include describe table permission, which is required when you hit the `/table` endpoint of the app.  It only grants read/write permissions, not describe or list.

Comment (sebsto) [reply lib/tictactoe-app-cdk-stack.ts]: Is there a way to add Describe* and List* permission without using FullAccess ? FullAccess also authorises to create and delete tables, index etc

Comment (rddefauw) [reply lib/tictactoe-app-cdk-stack.ts]: Fixed in 68869f4.

**Comment by rddefauw** [comment]
Hi, just checking to see if you're comfortable merging this into a v1 branch for now?  I can update it for v2 once v2 has support for all the cdk constructs I'm using.

**Comment by sebsto** [comment]
Sorry for the delay. I had to refactor files and directories to accommodate a change in the Route53 Application recovery Controller documentation.

The project and bootstrap is now [self hosted by the AWS Documentation team](https://r53-application-recovery-controller-cfn-app-iad.s3.us-east-1.amazonaws.com/README.html).

The new project structure has merge conflicts.  So I created a branch called "original" that was the last commit that you used for your modifications.

Can you change the pull request to propose it against the "original" branch instead of man ?  Then I will merge your changes.
Toc change the branch where to apply the PR, you can use the GitHub GUI, as per https://stackoverflow.com/questions/9135913/merge-pull-request-to-a-different-branch-than-default-in-github

**Comment by rddefauw** [comment]
Changed base branch to `original` per your note.

**Comment by sebsto** [comment]
Thank you ! Merged

**Pull request merged**